src/client.rs

```diff
@@ -1078,10 +1078,29 @@ impl<T: Read + Write> Session<T> {
     /// `EXISTS` response.  If the server does not do so, the client MAY issue a `NOOP` command (or
     /// failing that, a `CHECK` command) after one or more `APPEND` commands.
     pub fn append<S: AsRef<str>, B: AsRef<[u8]>>(&mut self, mailbox: S, content: B) -> Result<()> {
+        self.append_with_flags(mailbox, content, "()")
+    }
+
+    /// The [`APPEND` command](https://tools.ietf.org/html/rfc3501#section-6.3.11) can take
+    /// an optional FLAGS parameter to set the flags on the new message.
+    ///
+    /// > If a flag parenthesized list is specified, the flags SHOULD be set
+    /// > in the resulting message; otherwise, the flag list of the
+    /// > resulting message is set to empty by default.  In either case, the
+    /// > Recent flag is also set.
+    ///
+    /// Behaves like ['Session::append'], except takes a parenthesized list of flags.
```

Comment:
Can we just add the parentheses ourselves so the caller doesn't need to know about this detail?

Reply:
Haha! Yes! The only reason I did it this way is for consistency with the rest of the API. For example, `store` takes a flag list that is a space separated list of flag strings in parentheses, `status` takes `data_items` that is a space separated list in parentheses, etc. I am happy to be inconsistent to save the caller from needing to know this detail if you're happy with it.

Reply:
Hmm, that is a good point.. From memory, that might have been done to support the list being empty (and therefore absent) without having a different call. But here that seems like something we probably wouldn't want to support since the "flagless" version already exists as its own method. I think not requiring the parentheses is fine here.

As for the comma-separated part, I suppose you're right that that would be very inconsistent with `store`. That is unfortunate indeed. But at the same time, I think it's valuable to start pushing in the right direction where we have the option to without a breaking change. So if you can make it work with a well-typed list, I think that's preferable :)